be/src/olap/txn_manager.cpp

```diff
@@ -226,7 +226,7 @@ OLAPStatus TxnManager::publish_txn(OlapMeta* meta, TPartitionId partition_id, TT
         rowset_ptr->set_version_and_version_hash(version, version_hash);
         OLAPStatus save_status = RowsetMetaManager::save(meta,
             rowset_ptr->rowset_id(),
-            rowset_ptr->rowset_meta());
```

Comment:
Why not SharedPtr is not ok?

Reply:
because save api  just use pointer, no need to manage the object